packages/modules/src/SessionKeyManagerModule.ts

```diff
@@ -1,139 +1,203 @@
+import { generateRandomHex } from 'utils/UID'
 
-// Could be renamed with suffix API
 export class SessionKeyManagerModule extends BaseValidationModule {
-  // Review
-  sessionSigner!: Signer // optional global signer
-  sessionPubKey?: string // optional global public key
-  chainId: ChainId
-  moduleAddress!: string
+  chainId!: ChainId
   version: ModuleVersion = 'V1_0_0'
+  moduleAddress!: string
   nodeClient!: INodeClient
   merkleTree!: MerkleTree
-  // entryPoint!: EntryPoint
-
-  constructor(moduleConfig: SessionKeyManagerModuleConfig) {
+  sessionStorageClient!: SessionLocalStorage
+
+  /**
+   * This constructor is private. Use the static create method to instantiate SessionKeyManagerModule
+   * @param moduleConfig The configuration for the module
+   * @returns An instance of SessionKeyManagerModule
+   */
+  private constructor(moduleConfig: SessionKeyManagerModuleConfig) {
     super(moduleConfig)
+  }
+
+  /**
+   * Asynchronously creates and initializes an instance of SessionKeyManagerModule
+   * @param moduleConfig The configuration for the module
+   * @returns A Promise that resolves to an instance of SessionKeyManagerModule
+   */
+  public static async create(
+    moduleConfig: SessionKeyManagerModuleConfig
+  ): Promise<SessionKeyManagerModule> {
+    const instance = new SessionKeyManagerModule(moduleConfig)
+
     if (moduleConfig.moduleAddress) {
-      this.moduleAddress = moduleConfig.moduleAddress
+      instance.moduleAddress = moduleConfig.moduleAddress
     } else if (moduleConfig.version) {
       const moduleAddr = SESSION_MANAGER_MODULE_ADDRESSES_BY_VERSION[moduleConfig.version]
       if (!moduleAddr) {
         throw new Error(`Invalid version ${moduleConfig.version}`)
       }
-      this.moduleAddress = moduleAddr
-      this.version = moduleConfig.version as ModuleVersion
+      instance.moduleAddress = moduleAddr
+      instance.version = moduleConfig.version as ModuleVersion
     }
-    this.sessionSigner = moduleConfig.sessionSigner ?? ethers.Wallet.createRandom()
-    this.sessionPubKey = moduleConfig.sessionPubKey
-    this.chainId = moduleConfig.chainId
-    // this.entryPoint = ... // May not be needed at all
-    this.nodeClient = new NodeClient({
+    instance.chainId = moduleConfig.chainId
+    instance.nodeClient = new NodeClient({
       txServiceUrl: moduleConfig.nodeClientUrl ?? NODE_CLIENT_URL
     })
-    this.merkleTree = new MerkleTree([hexZeroPad('0x00', 32)], keccak256, { hashLeaves: false })
-  }
 
-  // Session Key Manager Module Address
-  getAddress(): string {
-    return this.moduleAddress
-  }
-
-  async getSigner(): Promise<Signer> {
-    throw new Error('Method not implemented.')
-  }
+    instance.sessionStorageClient = new SessionLocalStorage(moduleConfig.smartAccountAddress)
+
+    const existingSessionData = await instance.sessionStorageClient.getAllSessionData()
+    const existingSessionDataLeafs = existingSessionData.map((sessionData) => {
```

Comment:
Exclude the sessions whose validUltil date has passed or status is EXPIRED.

Reply:
I believe the merkle tree will still have the data its just it will be validate on chain? It should have all the data as per the discussion we had today

Reply:
Yeah as per the discussion, all the pending state sessions should be part of the merkle tree. Even though this recommended change doesn't affect the functionality in any way but it's just optimised way of doing things, where we can get rid of expired session leafs. 
Coz over time the leaves will keep on getting added and there's a limit of data that can be stored in local storage.

Not critical but we can add a TODO here to handle this later may be.